pvanalytics/quality/data_shifts.py

```diff
@@ -0,0 +1,298 @@
+    return
+
+
+def _erroneous_filter(time_series):
+    """
+    Remove any outliers from the time series.
+
+    Parameters
+    ----------
+    time_series : Pandas series with datetime index.
+        Daily time series of a PV data stream, which can include irradiance
+        and power data streams. This series represents the summed daily values
+        of the particular data stream.
+
+    Returns
+    -------
+    time_series: Pandas series, with a datetime index.
+        Time series, after filtering out outliers. This includes removal
+        of stale repeat readings, negative readings, and data greater than
+        the 99th percentile or less than the 1st percentile.
+    """
+    # Detect and mask stale data
+    stale_mask = gaps.stale_values_round(time_series, window=6,
+                                         decimals=3, mark='tail')
+    # Mask negative and 0 values
+    negative_mask = (time_series <= 0)
+    # Mask the top 1% and bottom 1% of data points
+    quantile_mask = ((time_series <= time_series.quantile(.01)) |
+                     (time_series >= time_series.quantile(.99)))
+    # Filter out the associated data by masking
+    time_series = time_series[(~stale_mask) & (~negative_mask) &
+                              (~quantile_mask)]
+    return time_series
+
+
+def _preprocess_data(time_series, remove_seasonality):
+    """
+    Pre-process the time series, including the following:
+        1. Min-max normalization of the time series.
+        2. Removing seasonality from the time series.
+
+    Parameters
+    ----------
+    time_series : Pandas series with datetime index.
+        Daily time series of a PV data stream, which can include irradiance
+        and power data streams. This series represents the summed daily values
+        of the particular data stream.
+    remove_seasonality: Boolean.
+        Whether or not to remove seasonality from the time series. If set to
+        True, the seasonality-removal routine is run on the min-max normalized
+        data. If not, the routine is skipped.
+
+    Returns
+    -------
+    Pandas series with a datetime index:
+        Time series, after data processing. This includes min-max
+        normalization, and, if the time series is in greater than 2 years in
+        length, seasonality removal.
+    """
+    # Min-max normalize the series
+    time_series_normalized = (time_series - time_series.min()) \
+        / (time_series.max() - time_series.min())
+    # Check if the time series is greater than two years in length. If not,
+    # pass back the normalized time series
+    if not remove_seasonality:
+        return time_series_normalized
+    else:
+        # Take the median of every day of the year across all years in the
+        # data, and use this as the seasonality of the time series
+        month_values = pd.DatetimeIndex(time_series.index).month
+        day_values = pd.DatetimeIndex(pd.Series(time_series.index)).day
```

Comment:
Good catch--no idea why this was here since we do an earlier check to ensure we have a time series index. Changed to time_series.index.month and time_series.index.day in https://github.com/pvlib/pvanalytics/commit/1f6c786e7511cd63d108759a78fdde3ff9217a1f